.github/workflows/ci.yml

```diff
@@ -68,3 +68,10 @@ jobs:
         with:
           working-directory: ./
           run: dub test -c unittest-gtk
+
+      - name: Test DWT Gtk (optimised)
+        if: ${{ matrix.os == 'ubuntu-20.04' && (matrix.dc == 'dmd-master' || matrix.dc == 'ldc-master') }}
```

Comment:
Hmm, perhaps better to add this to the matrix. The `if` becomes quite verbose.

Reply:
So I've taken a look at the documentation for [strategy.matrix](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix), but I'm not sure how to integrate the `if` into the matrix while keeping the `run`.

What may work is having something like the `experimental` key, as demonstrated in the [example including new combinations](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-including-new-combinations). Perhaps called `optimised`, and we'd check for that instead of what the `if` currently has.

If you had something else in mind, please let me know.

Reply:
> What may work is having something like the experimental key, as demonstrated in the example including new combinations

Yes, exactly. But I think you can skip the optimized step completely. You can add a new parameter to the matrix called `build`. The default value will be `unittest` (or what ever is the default build when running unit tests)  and you add two new entires to the matrix which sets the build to `optimised-unittest`. You end up with something like:

```yaml
os: [ubuntu-20.04, windows-2019]
build: unittest
dc:
  - dmd-master
  - ldc-master
  - dmd-beta
  - dmd-latest
  - ldc-latest
  - dmd-2.078.3
  - dmd-2.085.1
  - dmd-2.090.1
arch: [x86_64]
include:
  - os: windows-2019
    dc: dmd-latest
    arch: x86
  - os: windows-2019
    dc: ldc-latest
    arch: x86
  - os: ubuntu-20.04
    dc: ldc-latest
    build: optimized-unittest
  - os: ubuntu-20.04
    dc: dmd-latest
    build: optimized-unittest
```

```yaml
- name: Test DWT Gtk ${{ matrix.build }}
  if: ${{ matrix.os == 'ubuntu-20.04' }}
  uses: GabrielBB/xvfb-action@v1 # X virtual framebuffer
  with:
    working-directory: ./
    run: dub test -c unittest-gtk - ${{ matrix.build }}
```

Reply:
Think I got there in the end (had a few attempts :sweat_smile:). Let me know if there are any other issues.